pkg/codegen/internal/test/testdata/provider-config-schema/python/pulumi_configstation/config/vars.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# *** WARNING: this file was generated by test. ***
+# *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import warnings
+import pulumi
+import pulumi.runtime
+from typing import Any, Mapping, Optional, Sequence, Union, overload
+from .. import _utilities
+from . import outputs
+from .. import outputs as _root_outputs
+
+import types
+
+__config__ = pulumi.Config('configstation')
+
+
+class _ExportableConfig(types.ModuleType):
+    @property
+    def favorite_plants(self) -> Optional[Sequence[str]]:
+        return __config__.get('favoritePlants')
```

Comment:
I'm not sure if this is related to your changes, but this doesn't look right.

Since the type hint is `Optional[Sequence[str]]`, I would expect this to use `__config__.get_object` like the following (and possibly cast the result to make mypy happy, if needed):

```suggestion
    def favorite_plants(self) -> Optional[Sequence[str]]:
        return __config__.get_object('favoritePlants')
```

Looking at AWS, for Node.js you can see it's doing the right thing:

https://github.com/pulumi/pulumi-aws/blob/1b4fa4fecdc5b89c0141f07dcc9f65a4a1465d85/sdk/nodejs/config/vars.ts#L16

```typescript
export let allowedAccountIds: string[] | undefined = __config.getObject<string[]>("allowedAccountIds");
```

But for Python, it looks like we don't return the rich objects, we just always use `get` and return the string value (which for objects, will be a JSON string):

https://github.com/pulumi/pulumi-aws/blob/1b4fa4fecdc5b89c0141f07dcc9f65a4a1465d85/sdk/python/pulumi_aws/config/vars.py#L42

```python
allowed_account_ids = __config__.get('allowedAccountIds')
```

Unfortunately, I don't think we can change this without it being breaking for our existing providers. If we want to return actual objects, we'll probably want a Python setting in the schema so providers can opt-out (or opt-in depending on what we want the default behavior to be).

Or we just maintain the existing behavior and type all these as `Optional[str]`.

Reply:
Thanks for clarifying this, I think it makes sense to just type them all as Optional[str] for now, so we can maintain the current behavior consistently for automation API and multilang. 

I'll open up a follow-up issue that we can consider to return the rich objects and/or the output types.

I guess this begs the question, what's the point of the `config/outputs` at all if we're not using them anywhere? I imagine they are just being generated automatically as a result of the Input/Output typing work that you did last year but we're not actually using them... right?

Reply:
> I guess this begs the question, what's the point of the `config/outputs` at all if we're not using them anywhere? I imagine they are just being generated automatically as a result of the Input/Output typing work that you did last year but we're not actually using them... right?

Yeah, likely.

Reply:
To close the loop here, I have opted to maintain the status quo as fixing the types would involve a breaking change. I opened an issue to track returning the correct types: https://github.com/pulumi/pulumi/issues/7493